text/0005-application-formalization.md

```diff
@@ -0,0 +1,191 @@
+  "version": "0.0.1",
+  "description": "Application facilitating the placement of orders via GOBI and other compatible vendors",
+  "dependencies": [
+    {
+      "id": "app-orders-0.0.1",
+      "name": "app-orders",
+      "version": "^0.0.1"
+    },
+    {
+      "id": "app-organizations-0.0.1",
+      "name": "app-organizations",
+      "version": "^0.0.1"
+    }
+  ],
+  "modules": [
+    {
+      "id": "mod-gobi-2.6.0",
+      "name": "mod-gobi",
+      "version": "2.6.0",
+	  "url": "https://folio-registry.dev.folio.org/_/proxy/modules/mod-gobi-2.6.0"
+    }
+  ],
+  "uiModules": [
+    {
+      "id": "folio_gobi-settings-2.0.0",
+      "name": "folio_gobi-settings",
+      "version": "2.0.0",
+	  "url": "https://folio-registry.dev.folio.org/_/proxy/modules/folio_gobi-settings-2.0.0"
+    }
+  ]
+}
+```
+**N.B.** Full module descriptors and metadata have been omitted for readability reasons.
+
+## Benefits
+* Allows for expanding the bounded context
+  * By expanding the bounded context from the module to the Application, multiple modules can directly access the storage of the microservice.
+  * Benefites of this include:  
+    * Improved performance and simplified logic.  Modules can now perform the same operations with fewer API calls.
+    * Fewer modules since this eliminates the need for separate business logic and storage modules.
+* This is the first step toward breaking the monolithic release into application-specific releases
+* Results in fewer things to work with at the system operator level
+* Brings us closer to realizing the idea of having an application store/marketplace
+
+## Risks and Drawbacks
+* Folio has too many module dependencies which must be reduced
+* Transition to Applications will need to be iterative
+  * First pass is to create larger applications spanning multiple areas in order to satisfy dependencies
+  * Subsequent passes will further break up these large applications into smaller, applications focused on a single area of functionality
+* Since Applications are a vertical slice of functionality, better alignment/coordination between back-end and front-end development teams is required.
+  * Reorganizing development teams around applications would help
+* By latching onto the Application term, there may be confusion about terminology since that term currently means different things to different people.
+  * It may help to proposed formal names for other things currently referred to as Applications (e.g. the icons that appear in the Folio toolbar)
+
+## Rationale and Alternatives
+Alternative approaches include:
+* **Status quo** - Don't do anything and deal with monolithic releases, an ever-growing number of modules, and all the problems and challenges associated with those things.
+* **Independent module release cycles** - Impractical due to how tightly coupled modules are to one another.   It also doesn't address the issue of system operators needing to manage an ever growing number of modules.
```

Comment:
Let's resolve this in scope of the PRELIMINARY REVIEW stage since it's not about scope/general direction. I will address this in the DRAFT REFINEMENT stage.